atom/common/promise_util.h

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2018 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#ifndef ATOM_COMMON_PROMISE_UTIL_H_
+#define ATOM_COMMON_PROMISE_UTIL_H_
+
+#include <string>
+
+#include "native_mate/converter.h"
+
+namespace atom {
+
+namespace util {
+
+class Promise {
+ public:
+  explicit Promise(v8::Isolate* isolate) {
+    isolate_ = isolate;
+    resolver_.Reset(isolate, v8::Promise::Resolver::New(isolate));
+  }
+  ~Promise() { resolver_.Reset(); }
+
+  v8::Isolate* isolate() const { return isolate_; }
+
+  virtual v8::Local<v8::Object> GetHandle() const;
+
+  Promise* Resolve() {
+    GetInner()->Resolve(v8::Undefined(isolate()));
+    return this;
+  }
+
+  Promise* Reject() {
+    GetInner()->Reject(v8::Undefined(isolate()));
```

Comment:
The single-argument version of `Resolver::Reject()` is marked as `V8_DEPRECATE_SOON("Use maybe version"` ... would it be better to use the non-deprecated form here and below?